### PR TITLE
TreeBuilder build snippet usability improvements

### PR DESCRIPTION
**Pull request by tkvangorder**

### CODE REVIEW

File: rewrite-java/src/main/java/org/openrewrite/java/JavaParser.java [hunk 1]
```diff
@@ -57,6 +57,18 @@
                 .collect(toList());
     }
 
+    /**
```
Comment (tkvangorder) [inline rewrite-java/src/main/java/org/openrewrite/java/JavaParser.java:60]: Currently,` buildSnippet` was constructing a parser that was only aware of core Java library types. Introducing this method to help construct a parser that is aware of the entire runtime classpath.

File: rewrite-java/src/main/java/org/openrewrite/java/tree/MethodTypeBuilder.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+package org.openrewrite.java.tree;
+
+import com.koloboke.collect.set.hash.HashObjSet;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class MethodTypeBuilder {
```
Comment (tkvangorder) [inline rewrite-java/src/main/java/org/openrewrite/java/tree/MethodTypeBuilder.java:25]: I am still experimenting with a builder to make method type construction a little easier. I am open to any and all suggestions on where something like this should live?

**Review by jkschneider — APPROVED** [review]

File: rewrite-java/src/main/java/org/openrewrite/java/tree/MethodTypeBuilder.java (hunk 2 above)
Comment (jkschneider) [reply rewrite-java/src/main/java/org/openrewrite/java/tree/MethodTypeBuilder.java:25]: Having it alongside `TreeBuilder` seems reasonable to me.

Comment (jkschneider) [reply rewrite-java/src/main/java/org/openrewrite/java/tree/MethodTypeBuilder.java:25]: I'm sure at some point a "take this code and generate a tree out of it" interface that is language neutral.

**Pull request merged**